### PR TITLE
feat: onGenerated hook

### PR DESCRIPTION
**Pull request by schelmo**
Hello, i need access to the generated classes to save it for later use.
so i added an onGenerated option as callback. test is included.

### CODE REVIEW

File: packages/plugin-utils/src/options.ts [hunk 1]
```diff
@@ -120,6 +120,11 @@ export interface UserOptions {
    */
   sortUtilities?: boolean
 
+  /**
+   * Callback when classes and/or tags are generated/changed
+   */
+  onGenerated?: (classes: Set<string>, tags: Set<string>) => void
```
Comment (antfu) [inline packages/plugin-utils/src/options.ts:126]:
```suggestion
  onGenerated?: (ctx: { classes: Set<string>, tags: Set<string> }) => void
```

Would prefer passing as an object for extensibility in the future.

File: packages/plugin-utils/src/options.ts [hunk 2]
```diff
@@ -170,6 +175,7 @@ export interface ResolvedOptions {
   transformGroups: boolean
   sortUtilities: boolean
   safelist: Set<string>
+  onGenerated: Function
```
Comment (antfu) [inline packages/plugin-utils/src/options.ts:178]: Is this redundant?

Comment (schelmo) [reply packages/plugin-utils/src/options.ts:178]: im very new to typescript :)
without it, im getting: 
Property 'onGenerated' does not exist on type 'UserOptions | ResolvedOptions'

Comment (antfu) [reply packages/plugin-utils/src/options.ts:178]: I miss read that. Somehow the collapsing on GitHub is quite confusing:
image

No problem on this then.

**Pull request merged**